### ISSUE TITLE
Pattern recognition on y_testing output 

### ISSUE THREAD

**Issue by Southclaws** [Southclaws/sampctl#46]
A special flag should be added to a package runtime to specify whether the package is a test suite - this would give two things:

- Parsing of the output to give a nice output for editors/CI
- The ability to know when to kill the process when the tests are finished

**Comment by Southclaws** [comment]
sample of y_testing output:

```
Using cached package for 0.3.7
building /: with 3.10.4
Compiling source: 'F:\Projects\ScavengeSurviveOrg\mentions\test.pwn' with compiler 3.10.4...
Using cached package pawnc-3.10.4-windows.zip
- using include path F:\Projects\ScavengeSurviveOrg\mentions\dependencies\samp-stdlib
- using include path F:\Projects\ScavengeSurviveOrg\mentions\dependencies\amx_assembly
- using include path F:\Projects\ScavengeSurviveOrg\mentions\dependencies\YSI-Includes
- using include path F:\Projects\ScavengeSurviveOrg\mentions\dependencies\test-boilerplate
- using include path F:\Projects\ScavengeSurviveOrg\mentions\dependencies\ut_mock_players
Pawn compiler 3.10.4	 	 	Copyright (c) 1997-2006, ITB CompuPhase

Header size:           1472 bytes
Code size:           162180 bytes
Data size:           519360 bytes
Stack/heap size:      16384 bytes; estimated max. usage=2116 cells (8464 bytes)
Total requirements:  699396 bytes
start ./samp-server.exe in C:\Users\Southclaw\.samp\runtime\0.3.7

----------
Loaded log file: "server_log.txt".
----------

SA-MP Dedicated Server
----------------------
v0.3.7-R2, (C)2005-2015 SA-MP Team


Server Plugins
--------------
 Loaded 0 plugins.


Started server on port: 7777, with maxplayers: 50 lanmode is OFF.


Filterscripts
---------------
  Loaded 0 filterscripts.

 
 
 
 ======================================= 
 |                                     | 
 |        YSI version 4.00.0001        | 
 |        By Alex "Y_Less" Cole        | 
 |                                     | 
 ======================================= 
 
 
  ||===================||  
  || STARTING TESTS... ||  
  ||===================||  
 
*** Test ExpandMentions start
3: '{FF0000}red, {00FF00}green, {0000FF}blue.'
2: '{FF0000}red {00FF00}g'
1: '{000000}Player0{FFFFFF}'
3: 'Hello {000000}Player0{FFFFFF}, do you know {000000}Player1{FFFFFF} and {000000}Player2{FFFFFF}?'
3: 'attempt to recolour {FF0000}{000000}Player1{FFFFFF}{0000FF}'
 
PASS!
 
*** Test PopulationComplete start
Make sure that names are generated properly
Make sure that vehicles are copied properly
 
PASS!
 
*** Test y_testing_1 start
 
PASS!
 
*** Test y_testing_2 start
*** YSI Info: The next test hasn't really failed...
*** YSI Info:  - It is testing the test system.
FAIL: "false"
 
PASS!
 
*** Tests: 44, Fails: 0
 
  ||===================||  
  || ALL TESTS PASSED! ||  
  ||===================||  
 
Number of vehicle models: 0
--- Server Shutting Down.
```

**Comment by Southclaws** [comment]
regex:

`*** Test (\w+) start` $1 for test name

`PASS!` or `FAIL: (.*)` where $1 is test assertion that failed
and ending is just a summary
```
*** Tests: 52, Fails: 1
    - Test:MentionOne (1)
```
should subtract the total tests because of the y_testing ones

**Comment by Y-Less** [comment]
There's a reduced output mode:

```pawn
#define LIGHT_TEST_REPORT
```

It doesn't remove all the output, it just displays the end summary but no individual tests.

**Comment by Southclaws** [comment]
Ah, that will be useful, thanks!

**Comment by Southclaws** [comment]
Implemented in 1.6 release.
